akka-projection-slick/src/main/scala/akka/projection/slick/internal/SlickProjectionImpl.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.slick.internal
+
+import akka.Done
+import akka.actor.ClassicActorSystemProvider
+import akka.annotation.InternalApi
+import akka.event.Logging
+import akka.projection.Projection
+import akka.projection.slick.OffsetStore
+import akka.stream.scaladsl.{ Keep, Sink, Source }
+import akka.stream.{ KillSwitch, KillSwitches }
+import slick.basic.DatabaseConfig
+import slick.dbio.DBIO
+import slick.jdbc.JdbcProfile
+
+import scala.concurrent.Future
+@InternalApi
+private[projection] class SlickProjectionImpl[Offset, StreamElement, P <: JdbcProfile](
+    projectionId: String,
+    sourceProvider: Option[Offset] => Source[StreamElement, _],
+    offsetExtractor: StreamElement => Offset,
+    databaseConfig: DatabaseConfig[P],
+    eventHandler: StreamElement => DBIO[Done])
+    extends Projection {
+
+  private val offsetStore = new OffsetStore(databaseConfig.db, databaseConfig.profile)
+
+  private var shutdown: Option[KillSwitch] = None
+
+  override def start()(implicit systemProvider: ClassicActorSystemProvider): Unit = {
+
+    // TODO: add a LogSource for projection when we have a name and key
+    val akkaLogger = Logging(systemProvider.classicSystem, this.getClass)
+
+    implicit val dispatcher = systemProvider.classicSystem.dispatcher
+    import databaseConfig.profile.api._
+
+    val lastKnownOffset: Future[Option[Offset]] = offsetStore.readOffset(projectionId)
+
+    val killSwitch =
+      Source
+        .future(lastKnownOffset.map { offsetOpt =>
```

Comment:
Interesting, I think used `futureSource` in some previous version. I don't know why I changed anymore. 

We will need an execution context in the stop method (or have `def systemProvider: ClassicActorSystemProvider`) in order to flatMap it with the mat value (`Future[Done]`)

Reply:
I think `def systemProvider` makes sense. Constructor parameter.

Reply:
I'm in doubt about this. That means that a user will need it when constructing the Projection, while it's only used much later when we start if on their behalf. 

Both start/stop are not user functions, but internal API.

Reply:
I introduced a `ExecutionContext` to the `stop` method for the reasons I mention here. 

I didn't want to pass `ClassicActorSystemProvider` because we don't need an actor system, we need an EC.

Reply:
The start method is different, because then we need a `Materializer` so it justifies the `ClassicActorSystemProvider`

Reply:
Ok, makes sense as long as user api is not start/stop.